sdk/core/System.ClientModel/src/Pipeline/HttpClientPipelineTransport.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Primitives;
+
+public partial class HttpClientPipelineTransport : PipelineTransport, IDisposable
+{
+    /// <summary>
+    /// A shared instance of <see cref="HttpClientPipelineTransport"/> with default parameters.
+    /// </summary>
+    public static readonly HttpClientPipelineTransport Shared = new();
+
+    private readonly bool _ownsClient;
+    private readonly HttpClient _httpClient;
+
+    private bool _disposed;
+
+    public HttpClientPipelineTransport() : this(CreateDefaultClient())
+    {
+        // We will dispose the httpClient.
+        _ownsClient = true;
+    }
+
+    public HttpClientPipelineTransport(HttpClient client)
+    {
+        Argument.AssertNotNull(client, nameof(client));
+
+        _httpClient = client;
+
+        // The caller will dispose the httpClient.
+        _ownsClient = false;
+    }
+
+    private static HttpClient CreateDefaultClient()
+    {
+        // The following settings are added in Azure.Core and are not included
+        // in System.ClientModel.  If needed, we will migrate them to ClientModel
+        // over time.
+        //   - SSL settings
+        //   - Proxy settings
+        //   - Cookies
+        //   - MaxConnectionsPerServer
+        //   - PooledConnectionLifetime
+
+        HttpClientHandler handler = new HttpClientHandler()
+        {
+            AllowAutoRedirect = false
+        };
+
+        return new HttpClient(handler)
+        {
+            // Timeouts are handled by the pipeline
+            Timeout = Timeout.InfiniteTimeSpan,
+        };
+    }
+
+    protected override PipelineMessage CreateMessageCore()
+    {
+        PipelineRequest request = new HttpPipelineRequest();
+        PipelineMessage message = new PipelineMessage(request);
+
+        return message;
+    }
+
+    protected sealed override void ProcessCore(PipelineMessage message)
+    {
+#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult().
+
+#if NET6_0_OR_GREATER
+
+        ProcessSyncOrAsync(message, async: false).GetAwaiter().GetResult();
```

Comment:
is the `.EnsureCompleted` extension available here?

Reply:
I didn't pull it over before because of the demo timelines, but let me take a few minutes to see how hard that is to do.

Reply:
Addressed.